src/ranking_collecter/ranking_collector.py

```diff
@@ -0,0 +1,26 @@
+from string import Template
+import requests
+import user_info
+
+def collect_ranks(event, context):
+    usernames = user_info.USERNAMES
+    weekly_rankings = []
+    contest = user_info.CONTEST
+    weeklyContest = user_info.WEEKLYCONTEST
+
+    temp = Template('https://raw.githubusercontent.com/AWice/leetcode/master/leaderboard/scraped/${contestNo}weekly-contest-${weeklyContestNo}.txt')
+    url = temp.substitute(contestNo=contest, weeklyContestNo=weeklyContest)
+
+    github_request = requests.get(url = url)
+    data = github_request.text.split("\n")
+
+    for index, user in enumerate(data):
+        if user in usernames:
+            weekly_rankings.append(user)
+            usernames.remove(user)
+            if len(usernames) == 0:
+                break;
```

Comment:
this doesn't really improve runtime since you have to assume some of the people will be near the end of the list. if you must add it, I prefer something like

`if len(usernames) == len(weekly_rankings):`

Reply:
ignore the suggested if, didn't take into account that some users may not participate. Just remove